pkg/common/telemetry/prometheus.go

```diff
@@ -40,8 +40,16 @@ func newPrometheusRunner(c *MetricsConfig) (sinkRunner, error) {
 	}
 	handler := promhttp.HandlerFor(prometheus.DefaultGatherer, handlerOpts)
 
+	if runner.c.Host == "" {
+		runner.c.Host = "localhost"
+	}
+
+	if runner.c.Host != "localhost" {
```

Comment:
I didn't want to put this into an else branch of the above check because it would warn on an explicit `localhost` configuration. It will still warn on `127.0.0.0` etc. but checking for it would clutter the code unnecessarily, I think.